src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -200,6 +201,10 @@ var newlineEatingTags = {
   'textarea': true
 };
 
+var wrappers = {
+  'input': ReactDOMInput
```

Comment:
This pattern won't work if we want to support two passes of GCC advanced mode. Such as precompiling an npm package that someone else then compiles again.

In that scenario string properties means public API that someone else might provide. It does not mean that it will remain a string at runtime.